internal/nfs/controller/volume.go

```diff
@@ -169,22 +202,49 @@ func (nv *NFSVolume) DeleteExport() error {
 		return fmt.Errorf("failed to identify NFS cluster: %w", err)
 	}
 
-	// ceph nfs export rm <cluster_id> <pseudo_path>
-	args := []string{
-		"--id", nv.cr.ID,
-		"--keyfile=" + nv.cr.KeyFile,
-		"-m", nv.mons,
-		"nfs",
-		"export",
-		"delete",
-		nfsCluster,
-		nv.GetExportPath(),
+	// TODO: use new go-ceph API, see ceph/ceph-csi#2977
+	// new versions of Ceph use a different command, and the go-ceph API
+	// also seems to be different :-/
+	//
+	// run the new command, but fall back to the previous one in case of an
+	// error
+	cmds := [][]string{
+		// ceph nfs export rm <cluster_id> <pseudo_path>
+		{
+			"--id", nv.cr.ID,
+			"--keyfile=" + nv.cr.KeyFile,
+			"-m", nv.mons,
+			"nfs",
+			"export",
+			"rm",
+			nfsCluster,
+			nv.GetExportPath(),
+		},
+		// ceph nfs export delete <cluster_id> <pseudo_path>
+		{
+			"--id", nv.cr.ID,
+			"--keyfile=" + nv.cr.KeyFile,
+			"-m", nv.mons,
+			"nfs",
+			"export",
+			"delete",
```

Comment:
I think its good to have string constants for `create` and `delete` :

like `nfs export create cephfs` a single string..etc.

Reply:
I'll add a helper function instead